src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -0,0 +1,476 @@
+/**
+ * @file dbscan_test.cpp
+ * @author Nikhil Goel
+ *
+ * Test mlpackMain() of dbscan_main.cpp.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "DBSCAN";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/dbscan/dbscan_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct DBSCANTestFixture
+{
+ public:
+  DBSCANTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~DBSCANTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(DBSCANMainTest, DBSCANTestFixture);
+
+/**
+ * Check that number of output labels and number of input
+ * points are equal.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANOutputDimensionTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  size_t inputSize = inputData.n_cols;
+
+  SetInputParam("input", inputData);
+
+  mlpackMain();
+
+  // Check that number of predicted labels is equal to the input test points.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Row<size_t>>("assignments").n_cols,
+                      inputSize);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Row<size_t>>("assignments").n_rows,
+                      1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroids").n_rows, 2);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroids").n_cols, 4);
```

Comment:
May be you can write some small comments here about these numbers in centroids dimension check as it may vary with datasets. Your call?